src/compiler/llvm-to-backend/CMakeLists.txt

```diff
@@ -27,6 +27,10 @@ function(create_llvm_based_library)
     ${LLVM_TO_BACKEND_INCLUDE_DIRS})
   
   target_compile_definitions(${target} PRIVATE ${LLVM_DEFINITIONS} -DHIPSYCL_COMPILER_COMPONENT)
+  find_library(LLVM_LIBRARY NAMES LLVM HINTS ${LLVM_LIBRARY_DIRS})
+  if(NOT LLVM_LIBRARY)
+    message(FATAL_ERROR "LLVM at ${LLVM_DIR} does not have libLLVM.so. Please disable SSCP and OpenCL (-DWITH_SSCP_COMPILER=OFF -DWITH_OPENCL_BACKEND=OFF) or choose another LLVM installation")
```

Comment:
I think in principle you also need `-DWITH_LEVEL_ZERO_BACKEND=OFF`. The only reason why L0 does not cause this issue more frequently is because it is currently not auto-detected by cmake, unlike OpenCL. But if a user explicitly enables it, it would probably behave the same way.

(Independently of this PR, we should look into how we can resolve the dependency between SSCP <-> L0, OpenCL in a more intuitive, automatic way. Ideally `WITH_SSCP_COMPILER=OFF` would also disable depending backends)

Reply:
> I think in principle you also need -DWITH_LEVEL_ZERO_BACKEND=OFF. T

Good point, done.

> (Independently of this PR, we should look into how we can resolve the dependency between SSCP <-> L0, OpenCL in a more intuitive, automatic way. Ideally WITH_SSCP_COMPILER=OFF would also disable depending backends)

I fully agree, but my attempts at it produced something obscenely ugly :( I feel that, with so many feature flags, a proper dependency handling must be implemented, rather than manually coding a forest of `if()`s to handle all the possible combinations of manually set flags, automatically set flags, and environment factors while also providing helpful error messages.